### PR TITLE
Avoid losing some internal type information

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

File: packages/@ember/-internals/runtime/lib/type-of.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ const { toString } = Object.prototype;
 
   @method typeOf
   @for @ember/utils
-  @param {Object} item the item to check
+  @param item the item to check
```
Comment (NullVoxPopuli) [inline packages/@ember/-internals/runtime/lib/type-of.js:80]: what's the type if the type annotation is removed? `any`?

File: packages/@ember/-internals/glimmer/lib/helper.ts [hunk 2]
```diff
@@ -183,6 +187,10 @@ class ClassicHelperManager implements HelperManager<ClassicHelperStateBucket> {
   }
 }
 
+function isFactoryManager<T, C>(obj: unknown): obj is FactoryManager<T, C> {
+  return obj != null && 'class' in (obj as FactoryManager<T, C>);
```
Comment (NullVoxPopuli) [inline packages/@ember/-internals/glimmer/lib/helper.ts:191]: is a `.class` property the _only_ requirement for calling `create` on `obj`?

**Review by NullVoxPopuli — APPROVED** [review]

File: packages/@ember/-internals/runtime/lib/type-of.js (hunk 1 above)
Comment (wagenet) [reply packages/@ember/-internals/runtime/lib/type-of.js:80]: I believe so.

Comment (wagenet) [reply packages/@ember/-internals/runtime/lib/type-of.js:80]: It's either `any` or `unknown`.

File: packages/@ember/-internals/glimmer/lib/helper.ts (hunk 2 above)
Comment (wagenet) [reply packages/@ember/-internals/glimmer/lib/helper.ts:191]: https://github.com/emberjs/ember.js/blob/master/packages/@ember/-internals/container/lib/container.ts#L250-L253

Comment (NullVoxPopuli) [reply packages/@ember/-internals/glimmer/lib/helper.ts:191]: for extra certainty, should this function also check that create exists and is a function?

**Pull request merged**